projects/rates_example_test.go

```diff
@@ -0,0 +1,324 @@
+	if err != nil {
+		fmt.Printf("Error: %s\n", err)
+		return
+	}
+
+	fmt.Printf("Found %d user rates for project\n", len(resp.UserRates))
+
+	for _, userRate := range resp.UserRates {
+		fmt.Printf("User %d effective rate: %.2f\n", userRate.User.ID, userRate.EffectiveRate.Value())
+	}
+}
+
+func ExampleRateProjectUserGet() {
+	engine := twapi.NewEngine(session.NewBearerToken("your_token", "https://your-domain.teamwork.com"))
+
+	req := projects.NewRateProjectUserGetRequest(67890, 12345) // Project ID, User ID
+	req.Filters.Include = []string{"currencies"}
+
+	resp, err := projects.RateProjectUserGet(context.Background(), engine, req)
+	if err != nil {
+		fmt.Printf("Error: %s\n", err)
+		return
+	}
+
+	fmt.Printf("User rate for project: %d\n", resp.UserRate)
+	fmt.Printf("Rate value: %.2f\n", resp.Rate.Value())
+}
+
+func ExampleRateProjectUserUpdate() {
+	engine := twapi.NewEngine(session.NewBearerToken("your_token", "https://your-domain.teamwork.com"))
+
+	rate := int64(8500)
+	req := projects.NewRateProjectUserUpdateRequest(67890, 12345, &rate) // Project ID, User ID, Rate
+	resp, err := projects.RateProjectUserUpdate(context.Background(), engine, req)
+	if err != nil {
+		fmt.Printf("Error: %s\n", err)
+		return
+	}
+
+	fmt.Printf("Updated user rate to: %d\n", resp.UserRate)
+}
+
+func ExampleRateProjectUserHistoryGet() {
+	engine := twapi.NewEngine(session.NewBearerToken("your_token", "https://your-domain.teamwork.com"))
+
+	req := projects.NewRateProjectUserHistoryGetRequest(67890, 12345) // Project ID, User ID
+	req.Filters.OrderMode = "desc"                                    // Most recent first
+	req.Filters.PageSize = 10
+
+	resp, err := projects.RateProjectUserHistoryGet(context.Background(), engine, req)
+	if err != nil {
+		fmt.Printf("Error: %s\n", err)
+		return
+	}
+
+	fmt.Printf("Found %d historical rate entries\n", len(resp.UserRateHistory))
+
+	for _, history := range resp.UserRateHistory {
+		fmt.Printf("Rate: %.2f", history.Rate.Value())
+		if history.FromDate != nil {
+			fmt.Printf(" (effective from %s)", history.FromDate.Format("2006-01-02"))
+		}
+		if history.ToDate != nil {
+			fmt.Printf(" (until %s)", history.ToDate.Format("2006-01-02"))
+		}
+		fmt.Println()
+	}
+}
+
+// Example of working with pagination across all pages
+func ExampleRateProjectUserList_pagination() {
+	engine := twapi.NewEngine(session.NewBearerToken("your_token", "https://your-domain.teamwork.com"))
+
+	req := projects.NewRateProjectUserListRequest(67890)
+	req.Filters.PageSize = 50
+
+	allUserRates := []projects.EffectiveUserProjectRate{}
+
+	for {
+		resp, err := projects.RateProjectUserList(context.Background(), engine, req)
+		if err != nil {
+			fmt.Printf("Error: %s\n", err)
+			return
+		}
+
+		allUserRates = append(allUserRates, resp.UserRates...)
+
+		// Check if there are more pages
+		nextReq := resp.Iterate()
+		if nextReq == nil {
+			break
+		}
+		req = *nextReq
+	}
```

Comment:
Could we also apply the [iterator pattern](https://github.com/Teamwork/twapi-go-sdk/blob/2ff08ba1a6f1e0bcdac13b2d50b14145b78065b3/examples/iterator/iterator.go) here? It would avoid confusion with SDK users, showing a single way of doing it.